lib/router/src/lib.rs

```diff
@@ -102,37 +135,68 @@ impl<Req, Rec, Mk> Router<Req, Rec, Mk>
 where
     Rec: Recognize<Req>,
     Mk: Make<Rec::Target>,
-    Mk::Value: svc::Service<Req> + Clone,
+    <Mk as Make<Rec::Target>>::Value: Clone,
+    Mk::Value: svc::Service<Req>,
 {
-    pub fn new(recognize: Rec, make: Mk, capacity: usize, max_idle_age: Duration) -> Self {
-        Router {
-            inner: Arc::new(Inner {
+    pub fn new(
+        recognize: Rec,
+        make: Mk,
+        capacity: usize,
+        max_idle_age: Duration,
+    ) -> (Self, PurgeCache<Rec::Target, LoadShed<Mk::Value>>) {
+        let (cache, cache_bg) = Cache::new(capacity, max_idle_age);
+        let router = Self {
+            inner: Inner {
                 recognize,
                 make,
-                cache: Mutex::new(Cache::new(capacity, max_idle_age)),
-            }),
-        }
+                cache,
+            },
+        };
+
+        (router, cache_bg)
     }
 }
 
-impl<Req, Rec, Mk, Svc> svc::Service<Req> for Router<Req, Rec, Mk>
+impl<Req, Rec, Svc> Router<Req, Rec, FixedMake<Rec::Target, Svc>>
 where
     Rec: Recognize<Req>,
-    Mk: Make<Rec::Target, Value = Svc>,
     Svc: svc::Service<Req> + Clone,
-    Svc::Error: Into<error::Error>,
+{
+    /// A router that is created with a fixed set of known routes.
+    ///
+    /// `recognize` will only produce targets that exist in `routes`. We never
```

Comment:
Nit: I might change this to say
```suggestion
    /// `recognize` is expected to only produce targets that exist in `routes`. We never
```
we can't actually _enforce_ this; it's a contract that we expect the caller to adhere to.